src/hooks/sample/useSamplesWithQuestionaryStatus.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+
+import { SamplesFilter } from 'generated/sdk';
+import { useDataApi } from 'hooks/common/useDataApi';
+
+import { SampleWithQuestionaryStatus } from '../../models/Sample';
+
+export function useSamplesWithQuestionaryStatus(filter?: SamplesFilter) {
+  const [samples, setSamples] = useState<SampleWithQuestionaryStatus[]>([]);
+
+  const [samplesFilter, setSamplesFilter] = useState(filter);
+  const [loadingSamples, setLoadingSamples] = useState(false);
+  const api = useDataApi();
+
+  useEffect(() => {
```

Comment:
I suggest you add the check for unmounted component to avoid console warnings if component where the hook is used is destroyed before hook finished loading.